upgrade-and-test-plan.md

```diff
@@ -0,0 +1,68 @@
+## Steps to take to upgrade and test the library
+
+The proposal for introducing the following step is to adopt a git strategy as follows:
+
+1. a `dev` branch to merge against under the `hashicorp/next-remote-watch` repo.
+2. Every step of the following plan will be introduced as a separate PR to be merged against `dev` from our fork.
+3. Once all PRs under `dev` are merged, `dev` can be merged against `master/main` in the main repo.
```

Comment:
I don't really understand why we would need to do this. Can we just have a PR with the testing changes, and when that is done, we merge it? This way we don't need to deal with multiple PRs and a temporary branch.

Reply:
Also, this is not the appropriate place for such a document, as an actual file in the repository. These notes would be better kept elsewhere on your machine, or in a PR description perhaps. Ideally we can limit the code changes to actual code changes 🙏

Reply:
Yes absolutely, I will delete this file as it's not necessary considering I will be submitting all the changes into a single PR.

Will now close this then.